### PR TITLE
add VTK support for hybrid spaces

### PR DESCRIPTION
**Pull request by simonbyrne**
This reworks the internals a bit to support cell and point values, and modifies the frontend interface slightly.

### CODE REVIEW

**Review by jakebolewski — APPROVED** [review]
Looks good!  I was going to comment about supporting warped meshes for topo but I see there is already a TODO.

**Comment by jakebolewski** [comment]
We should add a way to get some viz assets from the VTK file output.

File: lib/ClimaCoreVTK/test/runtests.jl [hunk 1]
```diff
@@ -84,3 +84,72 @@ end
...
+
+    fspace = Spaces.ExtrudedFiniteDifferenceSpace(hspace, vspace)
+    cspace = Spaces.CenterExtrudedFiniteDifferenceSpace(fspace)
+    writevtk(
+        joinpath(dir, "hybrid2d_point"),
+        Fields.coordinate_field(fspace);
+        basis = :point,
+    )
+    writevtk(
+        joinpath(dir, "hybrid2d_cell"),
```
Comment (valeriabarra) [inline lib/ClimaCoreVTK/test/runtests.jl:151]: Should this have been called `hybrid3d_cell` and the one on line 146 `hybrid3d_point` ?

**Pull request merged**